lib/matplotlib/testing/compare.py

```diff
@@ -305,7 +306,16 @@ def convert(filename, cache):
             # re-generate any SVG test files using this mode, or else such tests will
             # fail to use the converter for the expected images (but will for the
             # results), and the tests will fail strangely.
-            if 'style="font:' in contents:
+            if re.search(
+                # searches for attributes :
+                #   style=[font|font-size|font-weight|
+                #          font-family|font-variant|font-style]
+                # taking care of the possibility of multiple style attributes
+                # before the font styling (i.e. opacity)
+                r'style="(?:\w+: [\w \'.]+; )*' +  # unrelated style attributes if any
+                r'font(?:|-size|-weight|-family|-variant|-style):',
```

Comment:
Can this be simplified to

```suggestion
                r'style="[^"]*font-(size|weight|family|variant|style):',
```
?
Reasoning: We're in the style definition as long as it is not terminated by ". I've also removed the `:?`. While we do not need the group, leaving out the non-matching group qualifier makes this easier to read.